### PR TITLE
`l2ImmutableDeployer` : remove contracts with no immutable

### PR DESCRIPTION
**Pull request by zhiqiangxu**
`l2ImmutableDeployer` is used to deploy contracts with immutables and [this line](https://github.com/ethereum-optimism/optimism/blob/b36eb5515cc2a34a15383b2eee488dbac83d6caf/op-chain-ops/immutables/immutables.go#L190) will stop contracts without immutables from being deployed.

This PR removes the contracts which `HasImmutableReferences` returns false.

### CODE REVIEW

**Comment by tynes** [comment]
Generally looks good to me, thank you for this fix

**Review by tynes — APPROVED** [review]

**Comment by zhiqiangxu** [comment]
@tynes I further removed some contracts with proxy disabled, the original branch is in effect noop since `codeAddr==predeploy.Address` which is already created in `setProxies`.

File: op-node/rollup/derive/ecotone_upgrade_transactions.go [hunk 1]
```diff
@@ -42,7 +42,7 @@ var (
 )
 
 func EcotoneNetworkUpgradeTransactions() ([]hexutil.Bytes, error) {
-	upgradeTxns := make([]hexutil.Bytes, 0, 5)
```
Comment (ajsutton) [inline op-node/rollup/derive/ecotone_upgrade_transactions.go:45]: Why did this get reduced? By my count there are 6 transactions being added so we should start with a capacity of 6.

Comment (zhiqiangxu) [reply op-node/rollup/derive/ecotone_upgrade_transactions.go:45]: I checked the history, it seems the [initial](cceb207a16) commit specified `5` and never changed afterwards. This PR changes it to `6`.

Comment (ajsutton) [reply op-node/rollup/derive/ecotone_upgrade_transactions.go:45]: oh my bad - I read the diff backwards.  Sorry.

**Review by ajsutton — APPROVED** [review]

File: op-node/rollup/derive/ecotone_upgrade_transactions.go (hunk 1 above)
Comment (zhiqiangxu) [reply op-node/rollup/derive/ecotone_upgrade_transactions.go:45]: never mind:)

**Pull request merged**